lib/rbs/definition_builder.rb

```diff
@@ -774,25 +780,13 @@ def merge_method(type_name, methods, name, method, sub, implemented_in: :keep, k
     end
 
     def try_cache(type_name, cache:, key: type_name)
-      # @type var cc: Hash[untyped, Definition | false | nil]
+      # @type var cc: Hash[untyped, Definition | nil]
       cc = _ = cache
-      cached = cc[key]
 
-      case cached
-      when Definition
-        cached
-      when false
-        raise
-      when nil
-        cc[key] = false
-        begin
-          cc[key] = yield
-        rescue => ex
-          cc.delete(key)
-          raise ex
-        end
+      if cc.locked?
+        cc[key] ||= yield
```

Comment:
This is a good improvement. 👏  The recursion check is moved away to `AncestorBuilder`, so we don't need assigning `false`.